test/CMakeLists.txt

```
@@ -41,3 +42,17 @@ target_link_libraries(testRpcClient
 )
 
 add_test("t-18-testRpcClient" ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/testRpcClient)
+
+add_executable(testPubSub
+	src/testPubSub.cpp)
+target_link_libraries(testPubSub
+	PUBLIC
+		up-client-zenoh-cpp::up-client-zenoh-cpp
+        spdlog::spdlog
+	PRIVATE
+		GTest::gtest_main
+		GTest::gmock    
+		pthread
+)
```

Minor: there's some indentation strangeness here.

```suggestion
add_executable(testPubSub
	src/testPubSub.cpp)
target_link_libraries(testPubSub
	PUBLIC
		up-client-zenoh-cpp::up-client-zenoh-cpp
		spdlog::spdlog
	PRIVATE
		GTest::gtest_main
		GTest::gmock    
		pthread
)
```

---

Please reach out to Mark Stokes